tests/integration/test_mongodb.py

```diff
@@ -53,6 +53,40 @@ class ExampleDto(ExampleCreationDto):
     id: str
 
 
+@pytest.mark.asyncio
+async def test_dao_find_all_with_id(mongodb_fixture: MongoDbFixture):  # noqa: F811
+    """Test using the id field as part of the mapping in find_all()"""
+
+    dao = await mongodb_fixture.dao_factory.get_dao(
+        name="example",
+        dto_model=ExampleDto,
+        dto_creation_model=ExampleCreationDto,
+        id_field="id",
+    )
+
+    # insert an example resource:
+    resource_to_create = ExampleCreationDto(field_a="test1", field_b=27, field_c=True)
+    resource_inserted = await dao.insert(resource_to_create)
+
+    assert isinstance(resource_inserted, ExampleDto)
+    assert resource_inserted.dict(exclude={"id"}) == resource_to_create.dict()
+
+    # retrieve the resource with find_all
+    resources_read = [
+        x async for x in dao.find_all(mapping={"id": resource_inserted.id})
+    ]
+    assert len(resources_read) == 1
+    assert resources_read[0].id == resource_inserted.id
+
+    # make sure the previous check wasn't a false positive
+    no_results = [x async for x in dao.find_all(mapping={"id": "noresults"})]
+    assert len(no_results) == 0
+
```

Comment:
Can you also add a test with an additional field like `field_b` to make sure that the other items in the mapping are not ignored when there is an id?

Reply:
See latest in [1d3bc90](https://github.com/ghga-de/hexkit/pull/63/commits/1d3bc90cea9723d460c96220ed1f1732ea9be06b)